hydradx/apps/money_market/toxic_debt.py

```diff
@@ -0,0 +1,281 @@
+    for tkn in mm.asset_list
+}
+
+with st.sidebar:
+    time_steps = st.number_input(
+        label="Time Steps",
+        min_value=1,
+        max_value=100,
+        value=10
+    )
+    price_factor = {
+        tkn: st.number_input(
+            label=f"{tkn} price change: ",
+            min_value=-99,
+            max_value=1000,
+            value=price_change_defaults[tkn],
+            key=tkn
+        ) for tkn in start_price
+    }
+
+# calculate price path from start price to crash price
+final_price = {tkn: start_price[tkn] * (1 + price_factor[tkn] / 100) for tkn in start_price}
+prices = {
+    tkn: [
+        start_price[tkn]] + [start_price[tkn] - (start_price[tkn] - final_price[tkn]) * ((i + 1) / time_steps)
+        for i in range(time_steps)
+    ] for tkn in start_price
+}
+full_trades = {tkn: [trade_to_price(omnipool, tkn, price) for price in prices[tkn]] for tkn in start_price}
+trade_sequence = [
+    [
+        {
+            'tkn_sell': tkn,
+            'tkn_buy': 'LRNA',
+            'sell_quantity': full_trades[tkn][i] - full_trades[tkn][i-1]
+        } if full_trades[tkn][i] < full_trades[tkn][i-1] else {
+            'tkn_sell': 'LRNA',
+            'tkn_buy': tkn,
+            'buy_quantity': full_trades[tkn][i - 1] - full_trades[tkn][i]
+        } for tkn in list(set(omnipool.asset_list) & set(mm.asset_list))
+    ] for i in range(1, time_steps + 1)
+]
+trade_sequence.extend([[
+    {'tkn_sell': tkn, 'tkn_buy': 'LRNA', 'sell_quantity': 0}
+    for tkn in list(set(omnipool.asset_list) & set(mm.asset_list))
+]])
+time_steps = len(trade_sequence)
+omnipool_sim = copy.deepcopy(omnipool)
+mm_sim = mm.copy()
+binance = FixedPriceExchange(
+    tokens={
+        tkn: omnipool.usd_price(tkn)
+        if tkn in omnipool.asset_list
+        else mm.price(tkn)
+        for tkn in list(set(omnipool.asset_list + mm.asset_list))
+    }, unique_id='binance'
+)
+
+config_list = [
+    {'exchanges': {'omnipool': ['DOT', 'vDOT'], '2-Pool-GDOT': ['aDOT', 'vDOT']}, 'buffer': 0.001},
+]
+
+initial_state = GlobalState(
+    pools=[router, mm_sim, omnipool_sim, binance, *stableswaps],
+    agents={
+        'liquidator': Agent(enforce_holdings=False, trade_strategy=liquidate_cdps('omnipool')),
+        'panic seller': Agent(
+            enforce_holdings=False,
+            trade_strategy=schedule_swaps(trade_sequence)
+        ),
+        'arbitrageur': Agent(
+            enforce_holdings=False,
+            trade_strategy=general_arbitrage(
+                exchanges=[omnipool, stableswaps[-1]],
+                config=config_list
+            )
+        )
+    },
+    evolve_function=update_prices
+)
+
+with st.spinner(f"Running {time_steps} simulation steps..."):
+    sim_start = time.time()
+    events = run(initial_state, time_steps=time_steps, silent=True)
+    sim_time = time.time() - sim_start
+    st.sidebar.info(f"Simulation completed in {sim_time:.2f} seconds")
+
+fig1, ax1 = plt.subplots()
```

Comment:
I'm wondering if it's worth adding a check that the price collapse that the user configures does in fact happen? Maybe displaying a warning to the user when it doesn't happen for some reason?
What I mean is, e.g. in the default config where the DOT price collapses, we are graphing the DOT price in USD over time. I can visually look and see that seems to match the 75% crash. Let's maybe add a real check that this is the case, and a warning to the user if it fails for some reason?